### PR TITLE
IMG-226 Adds support for "M" and "LU" in NitfRenderer getSourceBands

### PR DESCRIPTION
**Pull request by vinamartin**
https://codice.atlassian.net/browse/IMG-226
Adds in support for "M" and "LU". 
Comments are from the NITF 2.1 spec Table A-3. 

@dcruver @jlcsmith 
@clockard @emmberk @brendan-hofmann

### CODE REVIEW

File: render/src/main/java/org/codice/imaging/nitf/render/NitfRenderer.java [hunk 1]
```diff
@@ -202,6 +202,9 @@ private void renderJPEG2k(final ImageSegment imageSegment, final Graphics2D targ
     private int[] getSourceBands(final ImageSegment imageSegment) {
         List<Integer> imageBands = new ArrayList<Integer>();
 
+        /* When bands are marked as LU, R, G, B, and M are present
```
Comment (brendan-hofmann) [inline render/src/main/java/org/codice/imaging/nitf/render/NitfRenderer.java]: Think `When bands are` should just be `When bands`.

File: render/src/main/java/org/codice/imaging/nitf/render/NitfRenderer.java [hunk 2]
```diff
@@ -218,6 +221,44 @@ private void renderJPEG2k(final ImageSegment imageSegment, final Graphics2D targ
             }
         }
 
+        /* If R, G, B are not present, the default displayable band
+        is the LU band
+         */
+        if (imageBands.isEmpty()) {
```
Comment (clockard) [inline render/src/main/java/org/codice/imaging/nitf/render/NitfRenderer.java]: It looks like we do the same thing for LU and M bands as with RGB. Can we just add LU and M to the first switch statement with R,G and B?

Comment (bradh) [inline render/src/main/java/org/codice/imaging/nitf/render/NitfRenderer.java]: I think LU (if its needed) probably needs special case here.

Comment (vinamartin) [inline render/src/main/java/org/codice/imaging/nitf/render/NitfRenderer.java]: M and LU should be separate cases which I've tried to demonstrate in my comments. Hopefully that clears it up.

Comment (vinamartin) [inline render/src/main/java/org/codice/imaging/nitf/render/NitfRenderer.java]: This makes a lot more sense to me. Let me try this out and write a unit test to go along with it

**Comment by vinamartin** [comment]
Cleaned up code per @dcruver 's suggestion and added unit tests for that code per @brendan-hofmann 's suggestion.

File: render/src/test/java/org/codice/imaging/nitf/render/NitfRendererTest.java [hunk 3]
```diff
@@ -51,4 +89,43 @@ public void checkUnhandledCompressionException() throws IOException {
...
+    @Test
+    public void testMband() {
+        Mockito.when(mockImageSegmentHeader.getNumBands()).thenReturn(1);
+        Mockito.when(mockImageSegmentHeader.getImageBandZeroBase(0)).thenReturn(imageBandM);
+
+        int[] sourceBands = renderer.getSourceBands(mockImageSegmentHeader);
+        assertThat(sourceBands.length, is(1));
+        assertThat(sourceBands[0], is(0));
+    }
+
```
Comment (emmberk) [inline render/src/test/java/org/codice/imaging/nitf/render/NitfRendererTest.java]: Extra line

File: render/src/test/java/org/codice/imaging/nitf/render/NitfRendererTest.java [hunk 4]
```diff
@@ -51,4 +89,43 @@ public void checkUnhandledCompressionException() throws IOException {
         renderer.render(mockImageSegmentHeader, null);
     }
 
+
```
Comment (emmberk) [inline render/src/test/java/org/codice/imaging/nitf/render/NitfRendererTest.java]: Extra line

**Comment by bradh** [comment]
I'd still like to see it actually render.

**Comment by vinamartin** [comment]
@bradh I could probably do something like what's in the Jpeg2k test but for this case. Any suggestions?

**Comment by bradh** [comment]
@vinamartin I was expecting something like an extension to https://github.com/codice/imaging-nitf/blob/master/render/src/test/java/org/codice/imaging/nitf/render/RenderJitcTest.java

**Comment by bradh** [comment]
The render test also needs a reference image to check that it actually works like we expected. Render results end up in the `target` directory if the reference image is not found.  I've proposed a PR against your branch to fix that.

**Comment by vinamartin** [comment]
@bradh Thanks for the help, I've merged your PR to my branch.

**Pull request merged**